x/protocolpool/keeper/keeper.go

```diff
@@ -158,299 +119,87 @@ func (k Keeper) SetToDistribute(ctx sdk.Context) error {
 	}
 
 	// only take into account the balances of denoms whitelisted in EnabledDistributionDenoms
-	currentBalance := sdk.NewCoins()
+	amountToDistribute := sdk.NewCoins()
 	for _, denom := range params.EnabledDistributionDenoms {
 		bal := k.bankKeeper.GetBalance(ctx, moduleAccount.GetAddress(), denom)
-		currentBalance = currentBalance.Add(bal)
+		amountToDistribute = amountToDistribute.Add(bal)
 	}
```

Comment:
There was a PR that made a perf improvement for similar behavior here. maybe we can apply that? since we're getting the balance from distinct denoms, we can probably just append to `amountToDistribute`.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/24116/commits/8a8e465340cd2e861c759021aad16b4e2c435b89